In [33]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import classification_report
import pickle

In [85]:
DF=pd.read_excel('pesplanus_dataset1.xlsx')

In [ ]:
DF

In [86]:
train,valid,test=np.split(DF,[int(0.6*len(DF)),int(0.8*len(DF))])

In [87]:
def scale_dataset(DataFrame,oversample=False):
    x=DataFrame[DataFrame.columns[:-1]].values
    y=DataFrame[DataFrame.columns[-1]].values
    data=np.hstack((x, np.reshape(y,(-1,1))))
    return data,x,y

In [88]:
train,x_train,y_train=scale_dataset(train,oversample=True)
valid,x_valid,y_valid=scale_dataset(valid,oversample=False)
test,x_test,y_test=scale_dataset(test,oversample=False)

K-NearestNeighbors

In [89]:
knn_model=KNeighborsClassifier(n_neighbors=3)
knn_model.fit(x_train,y_train)


KNeighborsClassifier(n_neighbors=3)

In [90]:
y_pred=knn_model.predict(x_valid)

In [91]:
sum(abs(y_valid-y_pred))

10

In [92]:
classification_report(y_valid,y_pred)

'              precision    recall  f1-score   support\n\n           0       0.95      1.00      0.97       183\n           1       1.00      0.95      0.97       187\n\n    accuracy                           0.97       370\n   macro avg       0.97      0.97      0.97       370\nweighted avg       0.97      0.97      0.97       370\n'

            *K-NearestNeighbors*
              precision    recall  f1-score   support

           0       0.95      1.00      0.97       183
           1       1.00      0.95      0.97       187

    accuracy                           0.97       370
   macro avg       0.97      0.97      0.97       370
weighted avg       0.97      0.97      0.97       370


Support vector machine SVM

In [93]:
from sklearn.svm import SVC

In [94]:
svm_model=SVC()
svm_model.fit(x_train,y_train)
y_pred=svm_model.predict(x_valid)

In [95]:
sum(abs(y_valid-y_pred))

7

In [ ]:
classification_report(y_valid,y_pred)

            *Support vector machine*
              precision    recall  f1-score   support

           0       0.96      1.00      0.98       183
           1       1.00      0.96      0.98       187

    accuracy                           0.98       370
   macro avg       0.98      0.98      0.98       370
weighted avg       0.98      0.98      0.98       370


Logistic regression

In [101]:
from sklearn.linear_model import LogisticRegression

In [102]:
LG=LogisticRegression()
LG.fit(x_train,y_train)
y_pred=LG.predict(x_valid)


In [103]:
sum(abs(y_valid-y_pred))

187

In [ ]:
classification_report(y_valid,y_pred)

            *LogisticRegression*
              precision    recall  f1-score   support

           0       0.49      1.00      0.66       183
           1       0.00      0.00      0.00       187

    accuracy                           0.49       370
   macro avg       0.25      0.50      0.33       370
weighted avg       0.24      0.49      0.33       370


Random forest

In [105]:
from sklearn.ensemble import RandomForestClassifier

In [106]:
RF= RandomForestClassifier(max_depth=2, random_state=0)

In [107]:
RF.fit(x_train,y_train)

RandomForestClassifier(max_depth=2, random_state=0)

In [ ]:
RF.predict(x_valid)

In [ ]:
y_valid-y_pred

In [ ]:
classification_report(y_valid,y_pred)

              *RandomForest*
              precision    recall  f1-score   support

           0       0.49      1.00      0.66       183
           1       0.00      0.00      0.00       187

    accuracy                           0.49       370
   macro avg       0.25      0.50      0.33       370
weighted avg       0.24      0.49      0.33       370


NaiveBayes

In [111]:
from sklearn.naive_bayes import GaussianNB
NB=GaussianNB()
NB.fit(x_train,y_train)


GaussianNB()

In [112]:
y_pred=NB.predict(x_valid)

In [113]:
sum(abs(y_valid-y_pred))

11

In [ ]:
classification_report(y_valid,y_pred)

             *NaiveBayes*
              precision    recall  f1-score   support

           0       0.96      0.98      0.97       183
           1       0.98      0.96      0.97       187

    accuracy                           0.97       370
   macro avg       0.97      0.97      0.97       370
weighted avg       0.97      0.97      0.97       370
